lib/ui5Framework/Sapui5MavenSnapshotResolver.js

```diff
@@ -0,0 +1,195 @@
+		const {
+			cacheMode,
+			artifactsDir,
+			packagesDir,
+			metadataDir,
+			stagingDir,
+			sources,
+		} = options;
+
+		if (!snapshotEndpointUrl) {
+			snapshotEndpointUrl = Sapui5MavenSnapshotResolver.resolveSnapshotEndpointUrl();
+			// TODO: Save resolved snapshot URL when .ui5rc Configuration is implemented
+		}
+
+		this._installer = new Installer({
+			ui5HomeDir: this._ui5HomeDir,
+			snapshotEndpointUrl,
+			cacheMode,
+			artifactsDir,
+			packagesDir,
+			metadataDir,
+			stagingDir,
+		});
+		this._loadDistMetadata = null;
+		this._sources = !!sources;
+	}
+	loadDistMetadata() {
+		if (!this._loadDistMetadata) {
+			this._loadDistMetadata = Promise.resolve().then(async () => {
+				const version = this._version;
+				log.verbose(
+					`Installing ${DIST_ARTIFACT_ID} in version ${version}...`
+				);
+				const {pkgPath: distPkgPath} =
+					await this._installer.installPackage({
+						pkgName: DIST_PKG_NAME,
+						groupId: DIST_GROUP_ID,
+						artifactId: DIST_ARTIFACT_ID,
+						version,
+						classifier: "npm-sources",
+						extension: "zip",
+					});
+
+				return await this._installer.readJson(
+					path.join(distPkgPath, "metadata.json")
+				);
+			});
+		}
+		return this._loadDistMetadata;
+	}
+	async getLibraryMetadata(libraryName) {
+		const distMetadata = await this.loadDistMetadata();
+		const metadata = distMetadata.libraries[libraryName];
+
+		if (!metadata) {
+			throw new Error(`Could not find library "${libraryName}"`);
+		}
+
+		return metadata;
+	}
+	async handleLibrary(libraryName) {
+		const metadata = await this.getLibraryMetadata(libraryName);
+		if (!metadata.gav) {
+			throw new Error(
+				`Metadata is missing GAV information. This might indicate an unsupported SNAPSHOT version.`
+			);
+		}
+		const gav = metadata.gav.split(":");
+		let pkgName = metadata.npmPackageName;
+		if (!this._sources) {
+			pkgName += "-prebuilt";
+		}
+		return {
+			metadata: Promise.resolve({
+				id: pkgName,
+				version: metadata.version,
+				dependencies: metadata.dependencies,
+				optionalDependencies: metadata.optionalDependencies,
+			}),
+			// Trigger installation of package
+			install: this._installer.installPackage({
+				pkgName,
+				groupId: gav[0],
+				artifactId: gav[1],
+				version: metadata.version,
+				classifier: this._sources ? "npm-sources" : null,
+				extension: this._sources ? "zip" : "jar",
+			}),
+		};
+	}
+	static async fetchAllVersions({ui5HomeDir, cwd} = {}) {
+		const installer = new Installer({
+			cwd: cwd ? path.resolve(cwd) : process.cwd(),
+			ui5HomeDir: path.resolve(
+				ui5HomeDir || path.join(os.homedir(), ".ui5")
+			),
+		});
+		return await installer.fetchPackageVersions({
+			groupId: DIST_GROUP_ID,
+			artifactId: DIST_ARTIFACT_ID,
+		});
+	}
+	static async resolveSnapshotEndpointUrl(settingsXML, skipConfirmation) {
+		if (!skipConfirmation) {
+			const {default: yesno} = await import("yesno");
+			const ok = await yesno({
```

Comment:
I tried this locally via `ui5 serve` and it didn't throw or wait and then failed at some other place with `this._endpointUrl.endsWith is not a function`.

Reply:
Now should be fixed